synapse/lib/snap.py

```diff
@@ -210,13 +208,27 @@ def getNodesBy(self, full, valu=None, cmpr='='):
 
     def _getNodesByProp(self, full, valu=None, cmpr='='):
 
-        prop = self.model.prop(full)
+        prop: Optional[s_datamodel.Prop] = self.model.prop(full)
         if prop is None:
             raise s_exc.NoSuchProp(name=full)
 
         lops = prop.getLiftOps(valu, cmpr=cmpr)
-        for row, node in self.getLiftNodes(lops):
-            yield node
+
+        # We might actually lift nodes for which the comparison matches on lower layers, but doesn't on upper layers.
+        # We re-filter the nodes to catch these.
+
+        # FIXME: having to check type is awkward
+        if len(self.layers) > 1 and isinstance(prop, s_datamodel.Prop):
+            filt = prop.filt(text=valu, cmpr=cmpr)
+        else:
+            filt = None
+
+        # FIXME:  having to pack is awkward
```

Comment:
Agreed, the prop filter funcs pre-date the Node object, and should likely be updated to be aware of them.